### ISSUE TITLE
Detect usage of deprecated rules

### ISSUE THREAD

**Issue by matthieu-crouzet** [eslint/config-inspector#90]
It would be great to have a CLI to retrieve the list of deprecated rules activated (`warn`/`error`).
With that it would be easy to detect when some rules became deprecated while upgrading ESLint plugins versions.

The rule could be well deactivated if it is set to `off` for the same set or a superset of files compare to the set of files for which it is activated.
e.g.
```json5
[
  // Config coming from an external ESLint plugin
  {
    rules: {
      'my-deprecated-rule-1': 'warn',
      ...
    },
    files: ['**/*.{c,m,}{t,j}s']
  },
  // Local overrides
  {
     rules: {
       'my-deprecated-rule-1': 'off',
       ...
     },
    files: ['**/*']
  },
  {
     rules: {
       'my-deprecated-rule-2': 'off',
       ...
     },
    files: ['**/*.ts']
  }
]
```
In this case `my-deprecated-rule-1` is well deactivated but `my-deprecated-rule-2` is not because still activated for `['**/*.{c,m,}js', '**/*.{c,m}ts']`

It also could be great to detect "useless overloaded" rules
e.g.
```json5
[
  // Config coming from an external ESLint plugin
  {
    rules: {
      'my-rule': ['error', { configItem: 1 }]
    }
  },
  // Local override
  {
    rules: {
      'my-rule': ['error', { configItem: 1 }]
    }
  }
]
```
In this case I want to be aware that my local override is useless as it's already configured like that (only if the glob of the override is equal or a subset of the one defined in configuration from the plugin)

**Comment by ferferga** [comment]
Deprecated rules are shown straightaway in the UI:

![Image](https://github.com/user-attachments/assets/6eff37ec-98f6-4994-be6d-9f695d6fe478)

If you mean it in a CLI environment only, that is something with a bigger scope (I'd rename this issue to "Provide a CLI interface" in that case), since the inspector right now is just GUI-based, there's no CLI functionality.

Perhaps, it could even be said that it's out of the scope of this project and might fall better into upstream (eslint)?

**Comment by antfu** [comment]
Yeah, I think the point of having a UI inspector is that we would have complex requirements/viewports towards the config data, which CLI couldn't easily represent. Redoing the features of the inspector in CLI is considered out-of-scope of this project.
